tests/crypto/test_util.py

```diff
@@ -13,22 +15,24 @@
 
 
 class CryptoUtilTestCase(unittest.TestCase):
-    def setUp(self):
+    def setUp(self) -> None:
         super().setUp()
-        self.private_key = ec.generate_private_key(ec.SECP256K1(), default_backend())
+        self.private_key = cast(
+            EllipticCurvePrivateKeyWithSerialization, ec.generate_private_key(ec.SECP256K1(), default_backend())
+        )
```

Comment:
It does, but the function returns an `EllipticCurvePrivateKey`, while we expect an `EllipticCurvePrivateKeyWithSerialization`.

In the source, this looks like it's actually just an alias: https://github.com/pyca/cryptography/blob/main/src/cryptography/hazmat/primitives/asymmetric/ec.py#L125

But for some reason mypy complained about this, so I had to the cast. Maybe we could remove all mentions of `EllipticCurvePrivateKeyWithSerialization` from our code and simply use `EllipticCurvePrivateKey` everywhere.

Reply:
Yeah, I think it'd be better to refactor `EllipticCurvePrivateKeyWithSerialization` uses to `EllipticCurvePrivateKey` if that fixes any mypy complaints.

Reply:
I was wrong about this, they do have different methods. Which is weird considering the definition in the `ec.py` file linked above, but they're different in the `ec.pyi` file, and confirmed in the [docs](https://cryptography.io/en/3.3.2/hazmat/primitives/asymmetric/ec.html#cryptography.hazmat.primitives.asymmetric.ec.EllipticCurvePrivateKeyWithSerialization).

I changed it to use `assert isinstance` instead of a cast in https://github.com/HathorNetwork/hathor-core/pull/971/commits/2c62ee453e903da1d510e8b4c0858493dbc03631, which is also done in other parts of the code.